networkx/algorithms/tests/test_euler.py

```diff
@@ -80,18 +77,79 @@ def test_eulerian_circuit_digraph(self):
         assert_equal(nodes,[1,2,3,0])
         assert_equal(edges,[(1,2),(2,3),(3,0),(0,1)])
 
-
     def test_eulerian_circuit_multigraph(self):
         G=nx.MultiGraph()
         G.add_cycle([0,1,2,3])
         G.add_edge(1,2)
         G.add_edge(1,2)
         edges=list(eulerian_circuit(G,source=0))
-        nodes=[u for u,v in edges]
+        nodes=[u for u, v in edges]
         assert_equal(nodes,[0,3,2,1,2,1])
         assert_equal(edges,[(0,3),(3,2),(2,1),(1,2),(2,1),(1,0)])
 
+    @raises(nx.NetworkXError)
+    def test_not_eulerian(self):
+        next(list(eulerian_circuit(nx.complete_graph(4))))
+
+
+class TestEulerianPath:
+
+    # Use the bridges of Königsberg
+    bridges = nx.MultiGraph([('W', 'N'), ('W', 'N'), ('N', 'E'), ('E', 'W'),
+                             ('W', 'S'), ('W', 'S'), ('S', 'E')])
+
+    def verify_path(self, G, path):
```

Comment:
This function would be better if it had the assertions in its body. This way, failing tests indicate where they fail. For example,

    def verify_path(self, G, path):
        assert_not_equal(len(path), 0)
        assert_true(all(u in G and v in G for u, v in path))
        assert_true(all(G.has_edge(u, v) for u, v in path))
        assert_equal(set(G.edges()), set(path))